backend/app/controllers/api/v1/questions_controller.rb

```diff
@@ -0,0 +1,61 @@
+class Api::V1::QuestionsController < Api::ApiController
+
+  before_action :set_question, only: [:show, :update, :destroy, :answer]
+
+  def index
+    @questions =
+      if question_params[:difficulty_level]
+        Question.filter(question_params[:difficulty_level])
+      else
+        Question.all
+      end
+
+    case params[:version]
+    when "1"
+      json_response @questions
+    when "2"
+      json_response @questions.as_json(only: [:id, :answer])
+    end
```

Comment:
No need to check wether `params[:version]` is `1` or `2` because we already determine the versioning by module namespace explicitly.